mplleaflet/_display.py

```diff
@@ -139,8 +148,25 @@ def display(fig=None, closefig=True, **kwargs):
     if closefig:
         plt.close(fig)
 
-    html = fig_to_html(fig, template="ipynb.html", **kwargs)
-    return HTML(html)
+    html = fig_to_html(fig, **kwargs)
+
+    # We write explicitely the CSS and the JS into the html string.
+    html = html\
+        .replace('<link rel="stylesheet" href="http://cdn.leafletjs.com/leaflet-0.7.1/leaflet.css" />',
+                 '<style>{}</style>'.format(_leaflet_css))\
+        .replace('<script src="http://cdn.leafletjs.com/leaflet-0.7.1/leaflet.js"></script>',
+                 '<script>{}</script>'.format(_leaflet_js))
+
+    if iframe:
+        # We embed everything in an iframe.
+        iframe_html = '<iframe src="data:text/html;base64,{html}" width="{width}" height="{height}"></iframe>'\
+            .format(html = html.encode('base64'),
+                    width = int(60.*fig.get_figwidth()),
+                    height= int(60.*fig.get_figheight()),
+                   )
+        return HTML(iframe_html)
+    else:
+        return HTML(html)
```

Comment:
@BibMartin I couldn't get it working in the IPython notebook unless iframe was set to true. Otherwise we'd need the `ipynb.html` template. Is there any reason not to force iframe when calling display for the notebook here?

If not, probably makes sense to just remove the `iframe` argument and make that behavior the only part of the flow.